src/RA_StringUtils.h

```diff
@@ -45,9 +45,21 @@ _NODISCARD inline auto StringPrintf(_In_z_ _Printf_format_string_ const CharT* c
     }
     else if constexpr(std::is_same_v<CharT, wchar_t>)
     {
+        // starting capacity to reduce the need to reallocate, it could still be too small but is handled below
+        sFormatted.reserve(64U);
+        nNeeded = -1; // assume it already failed (usually does)
         assert(std::wstring_view{ sFormat }.find(L"%n") == std::wstring_view::npos);
         assert(sFormatted.capacity() > 0U && (sFormatted.capacity() < RSIZE_MAX/sizeof(wchar_t)));
-        nNeeded = std::vswprintf(sFormatted.data(), sFormatted.capacity(), sFormat, pArgs);
+
+        while(nNeeded < 0)
+        {
+            nNeeded = std::vswprintf(sFormatted.data(), sFormatted.capacity(), sFormat, pArgs);
+            if (nNeeded < 0)
+            {
+                const auto nCap = sFormatted.capacity();
+                sFormatted.reserve(nCap*2); // if it's still too small
+            }
+        }
```

Comment:
This change seems to be polluting all of your PRs. How hard is it to keep your branches separate?

Reply:
This change was originally done on my master branch. I felt it was important enough to apply it everywhere. 

If it's that important this not be here I can revert that commit from master and remove this fix from all my prs. It do you have a better idea?

Reply:
This is gone now, from all my PRs.

Reply:
Your master branch shouldn't have anything in it that's not in the main master branch. That's the point of having branches. I'd like you to rebase or squash your master to eliminate all the crap that's currently showing up in every one of your PRs:
![image](https://user-images.githubusercontent.com/32680403/48675168-f2c90880-eb12-11e8-9100-964225949d57.png)

Reply:
I'll do that